### ISSUE TITLE
Build fail on Theta if xmlchange'd CIME_OUTPUT_ROOT has trailing "/"

### ISSUE THREAD

**Issue by zarzycki** [ESMCI/cime#4321]
Weird build fail when building mct with E3SM (v2.0.0-1943-gceda11cfd0) using CIME (cime6.0.54-1-g98106c908) on Theta.

```
cd /home/zarzycki/E3SM/cime/scripts
./create_newcase --case /lus/grand/projects/ExtremeWeather/zarzycki/cases/TOM-F-betacast-F2010 --compset F2010-CICE --res ne30_ne30 --machine theta --compiler intel

cd $CASE
./xmlchange CIME_OUTPUT_ROOT=/lus/grand/projects/ExtremeWeather/zarzycki/

./case.setup
./case.build
```

fails during build with...

```
gmake -f /lus/grand/projects/ExtremeWeather/zarzycki/cases/TOM-F-betacast-F2010/Tools/Makefile  -C /lus/grand/projects/ExtremeWeather/zarzycki/TOM-F-betacast-F2010/bld/intel/mpt/nodebug/nothreads/mct/mct CIME_MODEL=e3sm  SMP=FALSE CASEROOT="/lus/grand/projects/ExtremeWeather/zarzycki/cases/TOM-F-betacast-F2010" CASETOOLS="/lus/grand/projects/ExtremeWeather/zarzycki/cases/TOM-F-betacast-F2010/Tools" CIMEROOT="/lus/swift/home/zarzycki/E3SM/cime" SRCROOT="/lus/swift/home/zarzycki/E3SM" COMP_INTERFACE="mct" COMPILER="intel" DEBUG="FALSE" EXEROOT="/lus/grand/projects/ExtremeWeather/zarzycki//TOM-F-betacast-F2010/bld" RUNDIR="/lus/grand/projects/ExtremeWeather/zarzycki//TOM-F-betacast-F2010/run" INCROOT="/lus/grand/projects/ExtremeWeather/zarzycki//TOM-F-betacast-F2010/bld/lib/include" LIBROOT="/lus/grand/projects/ExtremeWeather/zarzycki//TOM-F-betacast-F2010/bld/lib" MACH="theta" MPILIB="mpt" NINST_VALUE="c1a1l1i1o1r1g1w1i1e1" OS="CNL" PIO_VERSION=2 SHAREDLIBROOT="/lus/grand/projects/ExtremeWeather/zarzycki//TOM-F-betacast-F2010/bld" SMP_PRESENT="FALSE" USE_ESMF_LIB="FALSE" USE_MOAB="FALSE" CAM_CONFIG_OPTS="-mach theta -phys default -nlev 72 -clubb_sgs -microphys mg2 -chem linoz_mam4_resus_mom_soag -rain_evap_to_coarse_aero -bc_dep_to_snow_updates" COMP_LND="elm" COMPARE_TO_NUOPC="FALSE" USE_TRILINOS="FALSE" USE_ALBANY="FALSE" USE_PETSC="FALSE" COMP_NAME=mct /lus/grand/projects/ExtremeWeather/zarzycki/TOM-F-betacast-F2010/bld/intel/mpt/nodebug/nothreads/mct/mct/Makefile.conf
gmake: Entering directory '/lus/grand/projects/ExtremeWeather/zarzycki/TOM-F-betacast-F2010/bld/intel/mpt/nodebug/nothreads/mct/mct'
/lus/grand/projects/ExtremeWeather/zarzycki/cases/TOM-F-betacast-F2010/Tools/mkSrcfiles
cp -f Filepath Deppath
/lus/grand/projects/ExtremeWeather/zarzycki/cases/TOM-F-betacast-F2010/Tools/mkDepends  Deppath Srcfiles > Depends
gmake: Leaving directory '/lus/grand/projects/ExtremeWeather/zarzycki/TOM-F-betacast-F2010/bld/intel/mpt/nodebug/nothreads/mct/mct'
cat: Filepath: No such file or directory
cat: Srcfiles: No such file or directory
gmake: *** No rule to make target '/lus/grand/projects/ExtremeWeather/zarzycki/TOM-F-betacast-F2010/bld/intel/mpt/nodebug/nothreads/mct/mct/Makefile.conf'.  Stop.
ERROR: cat: Filepath: No such file or directory
cat: Srcfiles: No such file or directory
gmake: *** No rule to make target '/lus/grand/projects/ExtremeWeather/zarzycki/TOM-F-betacast-F2010/bld/intel/mpt/nodebug/nothreads/mct/mct/Makefile.conf'.  Stop.
```

However... running the exact same block with...

```
# Remove trailing backslash
./xmlchange CIME_OUTPUT_ROOT=/lus/grand/projects/ExtremeWeather/zarzycki
```

succeeds.

My best guess is something is getting dropped/added during a concatenation?

**Comment by jedwards4b** [comment]
I believe that #4305 fixes this problem.

**Comment by zarzycki** [comment]
Ack, I grepped GH for CIME_OUTPUT_ROOT and nothing of relevance appeared... I can verify that applying the fix in #4305 in my sandbox corrects! Closing.
